bofire/utils/torch_tools.py

```diff
@@ -85,6 +87,45 @@ def get_linear_constraints(
     return constraints
 
 
+def get_interpoint_constraints(
+    domain: Domain, n_candidates: int
+) -> List[Tuple[Tensor, Tensor, float]]:
+    """Converts interpoint equality constraints to linear equality constraints,
+        that can be processed by botorch. For more information, see the docstring
+        of `optimize_acqf` in botorch
+        (https://github.com/pytorch/botorch/blob/main/botorch/optim/optimize.py).
+
+    Args:
+        domain (Domain): Optimization problem definition.
+        n_candidates (int): Number of candidates that should be requested.
+
+    Returns:
+        List[Tuple[Tensor, Tensor, float]]: List of tuples, each tuple consists
+            of a tensor with the feature indices, coefficients and a float for the rhs.
+    """
+    constraints = []
+    for constraint in domain.constraints.get(InterpointEqualityConstraint):
+        assert isinstance(constraint, InterpointEqualityConstraint)
+        coefficients = torch.tensor([1.0, -1.0]).to(**tkwargs)
+        feat_idx = domain.get_feature_keys(Input).index(constraint.feature)
+        feat = domain.inputs.get_by_key(constraint.feature)
+        assert isinstance(feat, ContinuousInput)
+        if feat.is_fixed():
+            continue
+        multiplicity = constraint.multiplicity or n_candidates
+        for i in range(math.ceil(n_candidates / multiplicity)):
+            all_indices = torch.arange(
+                i * multiplicity, min((i + 1) * multiplicity, n_candidates)
+            )
+            for k in range(len(all_indices) - 1):
```

Comment:
One more thing, could you please explain to me why we need a pair of indices each time and not assign the feat_idx to each k entry, with k in range(len(all_indices))? thanks in advance

Reply:
Hmm, I am not sure that I understand you correct. We always need a pair of indices as we have several equalities of the form `feat_in_row_0 - feat-in_col_k == 0`. This makes two indices. And each index is by itself a pair because we have to index both the correct row and column index. 

Is this helping?